src/CaptureClient/CaptureEventProcessor.cpp

```diff
@@ -550,12 +552,32 @@ void CaptureEventProcessorForListener::ProcessThreadNamesSnapshot(
   }
 }
 
+ThreadStateSliceInfo::WakeupReason WakeupReasonConversion(
+    orbit_grpc_protos::ThreadStateSlice_WakeupReason reason) {
+  switch (reason) {
+    case orbit_grpc_protos::ThreadStateSlice_WakeupReason::
+        ThreadStateSlice_WakeupReason_kNotApplicable:
+      return ThreadStateSliceInfo::kNotApplicable;
+    case orbit_grpc_protos::ThreadStateSlice_WakeupReason::ThreadStateSlice_WakeupReason_kBlocked:
+      return ThreadStateSliceInfo::kBlocked;
+    case orbit_grpc_protos::ThreadStateSlice_WakeupReason::ThreadStateSlice_WakeupReason_kCreated:
+      return ThreadStateSliceInfo::kCreated;
+    default:
+      ORBIT_UNREACHABLE();
```

Comment:
I don't think this is the right default in this case. Maybe the service is sending an enum value that the client does not yet know about. In that case we should handle that gracefully and not crash. I would propose to return `kNotApplicable` in that case.

Reply:
I went with Davide's suggestion on this. Everything should have "not applicable" as a default value. If you reach this place then you initialized something wrong.